tests/test_io.py

```diff
@@ -253,6 +253,32 @@ def test_write_macro_block_size():
     assert frame_sizes[1] == (40, 50)
 
 
+def test_write_big_frames():
+    """Test that we give ffmpeg enough time to finish."""
+    try:
+        import numpy as np
+    except ImportError:
+        return skip("Missing 'numpy' test dependency")
+
+    sizes = []
+    for pixfmt, bpp in [("gray", 1), ("rgb24", 3), ("rgba", 4)]:
+        # Prepare for writing
+        gen = imageio_ffmpeg.write_frames(
+            test_file2, (2048, 2048), pix_fmt_in=pixfmt, ffmpeg_timeout=20.0
+        )
+        gen.send(None)  # seed
+        for i in range(9):
+            data = (255 * np.random.rand(2048 * 2048 * bpp)).astype(np.uint8)
+            data = bytes(data)
+            gen.send(data)
+        gen.close()
+        with open(test_file2, "rb") as f:
+            sizes.append(len(f.read()))
+        # Check nframes
+        nframes, nsecs = imageio_ffmpeg.count_frames_and_secs(test_file2)
+        assert nframes == 9
```

Comment:
Can you confirm that using the default 5s timeout will make this test fail? Maybe it's worth trying to set the timeout to 5, commit, push, see if CI fails?

Reply:
I could try calling it twice with the first one having the default timeout. I was also thinking about setting it to 1s to avoid a case where ffmpeg gets so fast that it can complete this faster than 5s. Let me try adding an expected fail first.

Reply:
As far as I can tell 5s should work... though it will be machine-dependent. What is the rationale for not going for a higher value? In the case of an actual hang of some kind, does it matter if the program waits 5 or 10s for example?

Reply:
> What is the rationale for not going for a higher value? In the case of an actual hang of some kind, does it matter if the program waits 5 or 10s for example?

Good point. I support it's mostly that I don't know what a good timeout would be. Should 10s be enough? or 20? Or maybe 50 in some cases? But if between us we've seen that 10s is enough, let's set it to 10 :)  I suspect that using a smaller buffer will lower the required timeout, but probably there will still be edge-cases, so this PR is worthwhile nonetheless.

Reply:
So as I'm preparing to test this, should this raise an exception instead of "silently" continuing on? If `ffmpeg` didn't close graciously, can't we assume it didn't create the file properly?

Reply:
...or does it fail with an exception? I feel like I've seen both.

Reply:
Another option would be to do `warnings.warn` instead of `logger.warning` so that it is easier to catch in a test.

Reply:
I would expect the assertion to fail, right? But maybe it's not that reproducible (i.e. sometimes ffmpeg might just be quick enough)?

Reply:
@almarklein Which assertion? Right now, the writing of frames only logs a warning if ffmpeg times out. If you try to read/use the video file it created then it is corrupt and not useable. That's why in my new version of the test I check for the ffmpeg process to fail when it *reads* the file. It never outwardly fails (warnings.warn, exception) for the write.

Reply:
I meant the `assert nframes == 9`, but I see what you mean now. Sounds good :)

Reply:
Unless you want me to change that log warning to a warnings.warn or exception, I think this PR is ready for merge based on everything else that was discussed. Let me know if you want anything else changed.